### ISSUE TITLE
Add references to MCNP Manual in docstrings for Input classes

### ISSUE THREAD

**Issue by tjlaboss** [idaholab/MontePy#413]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
For each publicly-available users manuals for versions of MCNP supported by MontePy, add a reference to the corresponding section of the user's manual where the input syntax is described. Right now, this is: [MCNP1](https://mcnp.lanl.gov/pdf_files/TechReport_1978_LANL_LA-7396-M_LGT.pdf), [MCNP2](https://mcnp.lanl.gov/pdf_files/TechReport_1979_LANL_LA-7396-MRev.1_LGT.pdf), [MCNP3A](https://mcnp.lanl.gov/pdf_files/TechReport_1986_LANL_LA-7396-MRev.2_BriesmeisterBoothEtAl.pdf), [MCNP4B](https://mcnp.lanl.gov/pdf_files/TechReport_1997_LANL_LA-12625-M_Briesmeisterothers.pdf), [MCNP4C](https://mcnp.lanl.gov/pdf_files/TechReport_2000_LANL_LA-13709-M_Briesmeisterothers.pdf), [MCNPX](https://mcnp.lanl.gov/pdf_files/TechReport_2002_LANL_LA-UR-02-2607_Waters.pdf), [MCNP5](https://mcnp.lanl.gov/pdf_files/TechReport_2003_LANL_LA-UR-03-1987Revised212008_SweezyBoothEtAl.pdf), [MCNP6.2](https://mcnp.lanl.gov/pdf_files/TechReport_2017_LANL_LA-UR-17-29981_WernerArmstrongEtAl.pdf), [MCNP6.3.0](https://mcnp.lanl.gov/pdf_files/TechReport_2022_LANL_LA-UR-22-30006Rev.1_KuleszaAdamsEtAl.pdf), [MCNP6.3.1](https://mcnp.lanl.gov/pdf_files/TechReport_2024_LANL_LA-UR-24-24602Rev.1_KuleszaAdamsEtAl.pdf), [MCNP6.3.2](https://mcnp.lanl.gov/pdf_files/TechReport_2026_LANL_LA-UR-26-23119Rev.1_KuleszaAdamsEtAl.pdf)

**Describe alternatives you've considered**
 - Only reference the latest (right now, MCNP6.3.1, specifically Chapter 5). This should work as the MCNP team strives to maintain backward compatibility.
 - Only reference manuals ≥MCNP6.2

**Additional context**
A good first issue for a new developer to become familiar with MontePy data types and MCNP syntax.

**Comment by MicahGale** [comment]
So 1. I didn't know about MCNP4C and MCNPX being publicly available.

As of right now we do support 6.2 and 6.3 exclusively officially. 

The 6.3 manual was moved all about so the section numbers are different. I want to add the 6.3 manual for sure. I'm not sure how I feel about 6.2. 

Some food for thought:

1. Can we reliably hyperlink to a specific section in any of the manuals? I feel like this should be possible.
2. Should we make a sphinx extension to automate this. 

```
.. mcnp:
   6.2: 5.1.2.3
   6.3: 1.2.3.4
```

This could then be rendered as:

Further reading:
| MCNP Version | Section    |
|--------------|------------|
| 6.2 | 5.1.2.3|
|6.3  | 1.2.3.4|

**Comment by tjlaboss** [comment]
Supporting only the latest would be cleanest. It would be the same amount of work to maintain this for only the latest release **or** all supported releases starting at 6.2.

Hyperlinking: I don't know how reliable this is, but you could do it. Here is an example based on a section I was just looking at.

 - Page link: [474](https://mcnp.lanl.gov/pdf_files/TechReport_2022_LANL_LA-UR-22-30006Rev.1_KuleszaAdamsEtAl.pdf#page=474)
 - Section link:  [5.9.14](https://mcnp.lanl.gov/pdf_files/TechReport_2022_LANL_LA-UR-22-30006Rev.1_KuleszaAdamsEtAl.pdf#subsection.5.9.14)

Obviously, the PDF link should be a variable so that it can be readily updated if the LANL website moves the PDFs again.

**Comment by MicahGale** [comment]
> Obviously, the PDF link should be a variable so that it can be readily updated ~if~ when the LANL website moves the PDFs again.

FTFY

**Comment by MicahGale** [comment]
I feel like we should split this into two tasks: build the machinery, and then populate the fields. I don't think building the machinery in sphinx would be necessarily a good first issue.

**Comment by MicahGale** [comment]
I'm wondering if there's an existing sphinx package that could be hacked to do this.

**Comment by tjlaboss** [comment]
https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html

**Comment by MicahGale** [comment]
Yes! We can also use that to link to issues from the changelog, etc!

**Comment by MicahGale** [comment]
related to #559.
